doc/modules/model_evaluation.rst

```diff
@@ -1378,6 +1379,36 @@ function::
     for an example of mean squared error usage to
     evaluate gradient boosting regression.
 
+.. _mean_squared_log_error:
+
+Mean squared log error
+----------------------
+
+The :func:`mean_squared_log_error` function computes a risk metric corresponding
+to the expected value of the logarithmic squared (quadratic) error or loss.
+
+If :math:`\hat{y}_i` is the predicted value of the :math:`i`-th sample,
+and :math:`y_i` is the corresponding true value, then the mean squared log error
+(MSLE) estimated over :math:`n_{\text{samples}}` is defined as
+
+.. math::
+
+  \text{MSLE}(y, \hat{y}) = \frac{1}{n_\text{samples}} \sum_{i=0}^{n_\text{samples} - 1} (\log (1 + y_i) - \log (1 +
```

Comment:
I presume this is meant to be applicable for non-negative regression targets? This should be stated. I think you should also give some sense of when this measure should be used, presumably for regressions over population counts and similar (i.e. targets with exponential growth).